core-bundle/src/EventListener/BackendCacheResponseListener.php

```diff
@@ -35,8 +35,12 @@ public function __invoke(ResponseEvent $event): void
         $request = $event->getRequest();
         $response = $event->getResponse();
 
+        // Requests with "Accept: text/vnd.turbo-stream.html, text/html" might return a
+        // different response body than "Accept: text/html" (#9128)
+        $response->headers->set('Vary', 'Accept');
```

Comment:
Thinking about it. We should vary on `X-Turbo-Request-Id`, no?

Reply:
I think the prefetch request should have the same request ID as the actual click-request. That would make sense to me.

Reply:
That's not the reason why this is there no. The 3s cache is there so that prefetch actually works. Otherwise hovering over a link (triggering prefetch) and then clicking would not take the prefetched request.
Hovering over multiple navigation items and re-using a cached entry is a different story - not even sure if we want this.

Reply:
Ah, I thought we had to do this only because of prefetch. Then yes, we have to do `accept` and `turbo-frame` to support both, frames and streams.